independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/QualifierConfigurator.java

```diff
@@ -0,0 +1,90 @@
+package io.quarkus.arc.processor;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.Type;
+import org.jboss.jandex.Type.Kind;
+
+/**
+ * This construct is not thread-safe.
```

Comment:
This will be user facing API, can we have some more verbose javadoc please?

Reply:
I can try to...

Reply:
@mkouba You know the drill - lorem ipsum ... :-D